driver-core/src/main/java/com/datastax/driver/core/policies/ErrorAwarePolicy.java

```diff
@@ -0,0 +1,529 @@
+     * Closes the policy. Calls the child policy's {@link LoadBalancingPolicy#close()} method, and shuts down the updater
+     * service created for the update of the {@link ErrorAwarePolicyTracker}.
+     */
+    @Override
+    public void close() {
+        childPolicy.close();
+        updaterService.shutdown();
+    }
+
+    /**
+     * Utility class to create a {@link ErrorAwarePolicy}.
+     */
+    public static class Builder {
+        final LoadBalancingPolicy childPolicy;
+
+        ErrorAwarePolicyTracker errorTracker;
+        long updatePeriod = 1000;
+        int errorsPerSecThreshold = 10;
+        int retryPeriod = 5000;
+
+        /**
+         * Creates a {@link Builder} instance.
+         *
+         * @param childPolicy the load balancing policy to wrap with error
+         *                    awareness.
+         */
+        public Builder(LoadBalancingPolicy childPolicy) {
+            this.childPolicy = childPolicy;
+        }
+
+        /**
+         * Defines the frequency at which the {@link ErrorAwarePolicyTracker#refresh} will be called.
+         * <p/>
+         * Default value for the update period is 1 second.
+         *
+         * @param updatePeriodMillis the number of milliseconds elapsed between each
+         * {@link ErrorAwarePolicyTracker#refresh} call.
+         * @return this {@link Builder} instance, for method chaining.
+         */
+        public Builder withUpdatePeriodMillis(long updatePeriodMillis) {
+            this.updatePeriod = updatePeriodMillis;
+            return this;
+        }
+
+        /**
+         * Defines the limit of errors per seconds after which a host will be considered INVALID
+         * for the {@link ErrorAwarePolicy}.
+         * <p/>
+         * The error tracker is going to get notified right away, by the {@link Cluster} it is registered to, of each error
+         * happening for each hosts. In addition to that the error tracker will get updated on a fixed period basis.
+         * During each of these updates, the {@link PerHostErrorTracker} can calculate for each host (thanks to the notifications
+         * from the cluster) a rate of errors per seconds based on the update frequency and the number of errors
+         * since the last update. The calculation will be done as such:
+         * <p/>
+         * Given :
+         * <ul>
+         *  <li>the number of errors happened since the last update \(nbErrors\),</li>
+         *  <li>the update frequency in milliseconds \(updatePeriod\) (see {@link #withUpdatePeriodMillis})</li>
+         *  <li>the rate 'milliseconds per seconds' \(millisToSec\) (= 1000)</li>
+         * </ul>
+         * <p/>
+         * The rate for each host \(errorsPerSecHostRate\) will be:
+         * \[
+         * errorsPerSecHostRate = \frac{nbErrors * updatePeriod}{millisToSec}
+         * \]
+         * <p/>
+         * This will give to the tracker an estimation of the number of errors/seconds currently happening for a host.
```

Comment:
During each `refresh()`, the errors in the `errors` map will only be the errors that happened since the last update. And **not** since the beginning, we reset the number of errors for each host in `refresh()`, see [here](https://github.com/datastax/java-driver/pull/705/files#diff-db1f1dde5efebe0746f5e8d06ab671a0R362).
If during a refresh, if the number of errors since the last `refresh()`, converted into errors/s (_errorRate_) overpasses the threshold, it triggers the mechanism to exclude the Host for a certain period of time (_retryPeriod_)